# Data Science Project

### by Mel Agatep
<br>
<br>
<br>

## Data Section <a name="data"></a>

#### Import necessary Libraries

In [2]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation

!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values

# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 
    
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize

!conda install -c conda-forge folium=0.5.0 --yes
import folium # plotting library

print('Folium installed')
print('Libraries imported.')

Solving environment: done

## Package Plan ##

  environment location: /home/jupyterlab/conda/envs/python

  added / updated specs:
    - geopy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    geographiclib-1.50         |             py_0          34 KB  conda-forge
    geopy-1.21.0               |             py_0          58 KB  conda-forge
    openssl-1.1.1g             |       h516909a_0         2.1 MB  conda-forge
    ------------------------------------------------------------
                                           Total:         2.2 MB

The following NEW packages will be INSTALLED:

  geographiclib      conda-forge/noarch::geographiclib-1.50-py_0
  geopy              conda-forge/noarch::geopy-1.21.0-py_0

The following packages will be UPDATED:

  openssl                                 1.1.1f-h516909a_0 --> 1.1.1g-h516909a_0



openssl-1.1.1g       | 2.1 MB    | ############

#### Define Foursquare Credentials and Version

In [3]:
CLIENT_ID = '2IFJ4OSQ4YRJ4ZF2BZZQXUNDJSNK4F4JHIFZ2RUVKVZ2FPQH' # your Foursquare ID
CLIENT_SECRET = 'FFJNCIUYMDQFICJLZMSZATWNHBBZCTCCZLRLUU1Z3JYSXNIS' # your Foursquare Secret
VERSION = '20180604'
LIMIT = 10
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: 2IFJ4OSQ4YRJ4ZF2BZZQXUNDJSNK4F4JHIFZ2RUVKVZ2FPQH
CLIENT_SECRET:FFJNCIUYMDQFICJLZMSZATWNHBBZCTCCZLRLUU1Z3JYSXNIS


In [4]:
# Use geopy library to get the latitude and longitude values of Downtown Toronto

address = 'Downtown Toronto, ON'
geolocator = Nominatim(user_agent="Downtown_Toronto_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Downtown Toronto are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Downtown Toronto are 43.6563221, -79.3809161.


<br>

### The Search for a specific venue

Imagine our targeted audience (Cuisine Connoisseurs, Food Enthusiast and Bloggers) is in a downtown area using the Restaurant Guide App and wanted to find great budget-friendly restaurants. It is a busy town, a quick comparison of several venues in close proximity would be ideal for they're always on the go. 
<br>

In [5]:
# define a query to search for Restaurant within 500 metres from Downtown Toronto

search_query = 'Restaurant'
radius = 500
print(search_query + ' .... OK!')

Restaurant .... OK!


#### Define the corresponding URL and limit = 10

In [6]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=2IFJ4OSQ4YRJ4ZF2BZZQXUNDJSNK4F4JHIFZ2RUVKVZ2FPQH&client_secret=FFJNCIUYMDQFICJLZMSZATWNHBBZCTCCZLRLUU1Z3JYSXNIS&ll=43.6563221,-79.3809161&v=20180604&query=Restaurant&radius=500&limit=10'

#### Make the call to foursquare api; send GET Request and examine results

In [7]:
results = requests.get(url).json()
results

# To simplify, output was cleared

{'meta': {'code': 200, 'requestId': '5eabcb92211536324bfcc0b2'},
 'response': {'venues': [{'id': '4ad4c05ff964a52048f720e3',
    'name': 'Hemispheres Restaurant & Bistro',
    'location': {'address': '110 Chestnut Street',
     'lat': 43.65488413420439,
     'lng': -79.38593077371578,
     'labeledLatLngs': [{'label': 'display',
       'lat': 43.65488413420439,
       'lng': -79.38593077371578}],
     'distance': 434,
     'postalCode': 'M5G 1R3',
     'cc': 'CA',
     'city': 'Toronto',
     'state': 'ON',
     'country': 'Canada',
     'formattedAddress': ['110 Chestnut Street',
      'Toronto ON M5G 1R3',
      'Canada']},
    'categories': [{'id': '4bf58dd8d48988d14e941735',
      'name': 'American Restaurant',
      'pluralName': 'American Restaurants',
      'shortName': 'American',
      'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_',
       'suffix': '.png'},
      'primary': True}],
    'referralId': 'v-1588317361',
    'hasPerk': False},
   {'id': '

#### Get relevant part of JSON and transform it into a *pandas* dataframe

In [8]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe.head()

/home/jupyterlab/conda/envs/python/lib/python3.6/site-packages/ipykernel_launcher.py:5: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead
  """


id                                name  \
0  4ad4c05ff964a52048f720e3     Hemispheres Restaurant & Bistro   
1  4edbccaaf9abeea3db5268df       Terrace Restaurant at Elmwood   
2  4fbbb3ace4b01e39077b4d55  Akashiro Japanese Restaurant & Bar   
3  53ae6660498e1ac3c1282743  Castries Bar & Five Way Restaurant   
4  4c6abb8c2c29d13ad2720e41                              Subway   

                                          categories    referralId  hasPerk  \
0  [{'id': '4bf58dd8d48988d14e941735', 'name': 'A...  v-1588317361    False   
1  [{'id': '4bf58dd8d48988d14e941735', 'name': 'A...  v-1588317361    False   
2  [{'id': '4bf58dd8d48988d1d2941735', 'name': 'S...  v-1588317361    False   
3  [{'id': '4bf58dd8d48988d144941735', 'name': 'C...  v-1588317361    False   
4  [{'id': '4bf58dd8d48988d1c5941735', 'name': 'S...  v-1588317361    False   

         location.address  location.lat  location.lng  \
0     110 Chestnut Street     43.654884    -79.385931   
1                     NaN     43.656101    -79.381480   
2           220 Yonge St.     43.655965    -79.380541   
3                     NaN     43.654728    -79.380928   
4  10 Dundas St E, No 337     43.656494    -79.380798   

                             location.labeledLatLngs  location.distance  \
0  [{'label': 'display', 'lat': 43.65488413420439...                434   
1  [{'label': 'display', 'lat': 43.65610066909799...                 51   
2  [{'label': 'display', 'lat': 43.65596504293232...                 49   
3  [{'label': 'display', 'lat': 43.65472793579101...                177   
4  [{'label': 'display', 'lat': 43.6564939, 'lng'...                 21   

  location.postalCode location.cc location.city location.state  \
0             M5G 1R3          CA       Toronto             ON   
1                 NaN          CA           NaN            NaN   
2                 NaN          CA       Toronto             ON   
3                 NaN          CA           NaN            NaN   
4             M5B 2G9          CA       Toronto             ON   

  location.country                          location.formattedAddress  \
0           Canada  [110 Chestnut Street, Toronto ON M5G 1R3, Canada]   
1           Canada                                           [Canada]   
2           Canada  [220 Yonge St. (in Toronto Eaton Centre Urban ...   
3           Canada                                           [Canada]   
4           Canada  [10 Dundas St E, No 337 (at Yonge St.), Toront...   

                   location.crossStreet venuePage.id  
0                                   NaN          NaN  
1                                   NaN          NaN  
2  in Toronto Eaton Centre Urban Eatery          NaN  
3                                   NaN          NaN  
4                          at Yonge St.          NaN

#### Define information of interest and filter dataframe

In [9]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name            categories  \
0                Hemispheres Restaurant & Bistro   American Restaurant   
1                  Terrace Restaurant at Elmwood   American Restaurant   
2             Akashiro Japanese Restaurant & Bar      Sushi Restaurant   
3             Castries Bar & Five Way Restaurant  Caribbean Restaurant   
4                                         Subway        Sandwich Place   
5  Spring Rolls | Japanese Restaurant in Toronto      Theme Restaurant   
6                        Joe's Indian Restaurant     Indian Restaurant   
7                     Alio Restaurant & Wine Bar    Italian Restaurant   
8                               Adega Restaurant            Restaurant   
9                  Hong Shing Chinese Restaurant    Chinese Restaurant   

                  address        lat        lng  \
0     110 Chestnut Street  43.654884 -79.385931   
1                     NaN  43.656101 -79.381480   
2           220 Yonge St.  43.655965 -79.380541   
3                     NaN  43.654728 -79.380928   
4  10 Dundas St E, No 337  43.656494 -79.380798   
5  40 Dundas St W Box 36,  43.656105 -79.383495   
6                     NaN  43.658150 -79.381563   
7         108 Dundas St W  43.655655 -79.384124   
8               33 Elm St  43.657519 -79.383462   
9         195 Dundas St W  43.654925 -79.387089   

                                      labeledLatLngs  distance postalCode  cc  \
0  [{'label': 'display', 'lat': 43.65488413420439...       434    M5G 1R3  CA   
1  [{'label': 'display', 'lat': 43.65610066909799...        51        NaN  CA   
2  [{'label': 'display', 'lat': 43.65596504293232...        49        NaN  CA   
3  [{'label': 'display', 'lat': 43.65472793579101...       177        NaN  CA   
4  [{'label': 'display', 'lat': 43.6564939, 'lng'...        21    M5B 2G9  CA   
5  [{'label': 'display', 'lat': 43.65610515821675...       209    M5G 2C2  CA   
6  [{'label': 'display', 'lat': 43.65814977325445...       210        NaN  CA   
7  [{'label': 'display', 'lat': 43.65565507065502...       268    M5G 1C3  CA   
8  [{'label': 'display', 'lat': 43.65751886446546...       244    M5G 1H1  CA   
9  [{'label': 'display', 'lat': 43.65492521335936...       520    M5G 1C7  CA   

      city state country                                   formattedAddress  \
0  Toronto    ON  Canada  [110 Chestnut Street, Toronto ON M5G 1R3, Canada]   
1      NaN   NaN  Canada                                           [Canada]   
2  Toronto    ON  Canada  [220 Yonge St. (in Toronto Eaton Centre Urban ...   
3      NaN   NaN  Canada                                           [Canada]   
4  Toronto    ON  Canada  [10 Dundas St E, No 337 (at Yonge St.), Toront...   
5  Toronto    ON  Canada  [40 Dundas St W Box 36,, Toronto ON M5G 2C2, C...   
6      NaN   NaN  Canada                                           [Canada]   
7  Toronto    ON  Canada  [108 Dundas St W (at Bay St.), Toronto ON M5G ...   
8  Toronto    ON  Canada  [33 Elm St (at Bay St), Toronto ON M5G 1H1, Ca...   
9  Toronto    ON  Canada  [195 Dundas St W (at University Ave), Toronto ...   

                            crossStreet                        id  
0                                   NaN  4ad4c05ff964a52048f720e3  
1                                   NaN  4edbccaaf9abeea3db5268df  
2  in Toronto Eaton Centre Urban Eatery  4fbbb3ace4b01e39077b4d55  
3                                   NaN  53ae6660498e1ac3c1282743  
4                          at Yonge St.  4c6abb8c2c29d13ad2720e41  
5                                   NaN  56dd9d68498eb4e5edcb30f9  
6                                   NaN  5165c333e4b07a7ad88d8a69  
7                            at Bay St.  4ae9ef82f964a520cbb721e3  
8                             at Bay St  4ad4c060f964a52070f720e3  
9                     at University Ave  4b2027b5f964a520f82d24e3

In [10]:
dataframe_filtered.name

0                  Hemispheres Restaurant & Bistro
1                    Terrace Restaurant at Elmwood
2               Akashiro Japanese Restaurant & Bar
3               Castries Bar & Five Way Restaurant
4                                           Subway
5    Spring Rolls | Japanese Restaurant in Toronto
6                          Joe's Indian Restaurant
7                       Alio Restaurant & Wine Bar
8                                 Adega Restaurant
9                    Hong Shing Chinese Restaurant
Name: name, dtype: object

<br>

### Map of The City’s Downtown Area 

Would it be convenient if we could show our targeted audience the map of their search. Also this will show them how far they are from a prospective venue.

In [11]:
# Let’s locate the approximate center and explore neighboring restaurants
# Zoom in to visualize
# Note: Github may not display map

venues_map = folium.Map(location=[latitude, longitude], zoom_start=10) # generate map centred around 

# add a red circle marker to represent 
folium.features.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Downtown Toronto',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)

# add the restaurants as blue circle markers
for lat, lng, label in zip(dataframe_filtered.lat, dataframe_filtered.lng, dataframe_filtered.categories):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map

##### Approximate Defined Center: Google Maps

We searched cordinate 43.6563221, -79.3809161 for Downtown Toronto in google maps. Result will shows "418 The PATH - Yonge-Dundas Square, Toronto, ON M5B 1R4, Canada". This is our approximate defined center and represented by a red circle marker from the generated map above. Using the result, we can test venue location accuracy and other information.

#### Dataframe Clean Up

In [12]:
# Let's sort by distance
sort_by_distance = dataframe_filtered.sort_values('distance')
sort_by_distance

name            categories  \
4                                         Subway        Sandwich Place   
2             Akashiro Japanese Restaurant & Bar      Sushi Restaurant   
1                  Terrace Restaurant at Elmwood   American Restaurant   
3             Castries Bar & Five Way Restaurant  Caribbean Restaurant   
5  Spring Rolls | Japanese Restaurant in Toronto      Theme Restaurant   
6                        Joe's Indian Restaurant     Indian Restaurant   
8                               Adega Restaurant            Restaurant   
7                     Alio Restaurant & Wine Bar    Italian Restaurant   
0                Hemispheres Restaurant & Bistro   American Restaurant   
9                  Hong Shing Chinese Restaurant    Chinese Restaurant   

                  address        lat        lng  \
4  10 Dundas St E, No 337  43.656494 -79.380798   
2           220 Yonge St.  43.655965 -79.380541   
1                     NaN  43.656101 -79.381480   
3                     NaN  43.654728 -79.380928   
5  40 Dundas St W Box 36,  43.656105 -79.383495   
6                     NaN  43.658150 -79.381563   
8               33 Elm St  43.657519 -79.383462   
7         108 Dundas St W  43.655655 -79.384124   
0     110 Chestnut Street  43.654884 -79.385931   
9         195 Dundas St W  43.654925 -79.387089   

                                      labeledLatLngs  distance postalCode  cc  \
4  [{'label': 'display', 'lat': 43.6564939, 'lng'...        21    M5B 2G9  CA   
2  [{'label': 'display', 'lat': 43.65596504293232...        49        NaN  CA   
1  [{'label': 'display', 'lat': 43.65610066909799...        51        NaN  CA   
3  [{'label': 'display', 'lat': 43.65472793579101...       177        NaN  CA   
5  [{'label': 'display', 'lat': 43.65610515821675...       209    M5G 2C2  CA   
6  [{'label': 'display', 'lat': 43.65814977325445...       210        NaN  CA   
8  [{'label': 'display', 'lat': 43.65751886446546...       244    M5G 1H1  CA   
7  [{'label': 'display', 'lat': 43.65565507065502...       268    M5G 1C3  CA   
0  [{'label': 'display', 'lat': 43.65488413420439...       434    M5G 1R3  CA   
9  [{'label': 'display', 'lat': 43.65492521335936...       520    M5G 1C7  CA   

      city state country                                   formattedAddress  \
4  Toronto    ON  Canada  [10 Dundas St E, No 337 (at Yonge St.), Toront...   
2  Toronto    ON  Canada  [220 Yonge St. (in Toronto Eaton Centre Urban ...   
1      NaN   NaN  Canada                                           [Canada]   
3      NaN   NaN  Canada                                           [Canada]   
5  Toronto    ON  Canada  [40 Dundas St W Box 36,, Toronto ON M5G 2C2, C...   
6      NaN   NaN  Canada                                           [Canada]   
8  Toronto    ON  Canada  [33 Elm St (at Bay St), Toronto ON M5G 1H1, Ca...   
7  Toronto    ON  Canada  [108 Dundas St W (at Bay St.), Toronto ON M5G ...   
0  Toronto    ON  Canada  [110 Chestnut Street, Toronto ON M5G 1R3, Canada]   
9  Toronto    ON  Canada  [195 Dundas St W (at University Ave), Toronto ...   

                            crossStreet                        id  
4                          at Yonge St.  4c6abb8c2c29d13ad2720e41  
2  in Toronto Eaton Centre Urban Eatery  4fbbb3ace4b01e39077b4d55  
1                                   NaN  4edbccaaf9abeea3db5268df  
3                                   NaN  53ae6660498e1ac3c1282743  
5                                   NaN  56dd9d68498eb4e5edcb30f9  
6                                   NaN  5165c333e4b07a7ad88d8a69  
8                             at Bay St  4ad4c060f964a52070f720e3  
7                            at Bay St.  4ae9ef82f964a520cbb721e3  
0                                   NaN  4ad4c05ff964a52048f720e3  
9                     at University Ave  4b2027b5f964a520f82d24e3

In [13]:
# Hide the index
sort_by_distance.style.hide_index()

In [14]:
pip install openpyxl

     |████████████████████████████████| 174kB 5.9MB/s eta 0:00:01
  Stored in directory: /home/jupyterlab/.cache/pip/wheels/b5/85/ca/e768ac132e57e75e645a151f8badac71cc0089e7225dddf76b
  Stored in directory: /home/jupyterlab/.cache/pip/wheels/2a/77/35/0da0965a057698121fc7d8c5a7a9955cdbfb3cc4e2423cad39
Successfully built openpyxl et-xmlfile
Note: you may need to restart the kernel to use updated packages.


In [15]:
sort_by_distance.to_excel ('tableu.xlsx')

#### Explore Restaurants using their ID

In [16]:
# Making calls to Foursquare API to get 'given values'
# We then get the Rating, Price, Tier and Currency for others

In [17]:
venue_id = '4ad7929cf964a520500c21e3' # ID of The Senator Restaurant
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4ad7929cf964a520500c21e3?client_id=2IFJ4OSQ4YRJ4ZF2BZZQXUNDJSNK4F4JHIFZ2RUVKVZ2FPQH&client_secret=FFJNCIUYMDQFICJLZMSZATWNHBBZCTCCZLRLUU1Z3JYSXNIS&v=20180604'

In [18]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

# To simplify, output was cleared

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'venuePage', 'reasons', 'description', 'storeId', 'page', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'hours', 'popular', 'seasonalHours', 'defaultHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '4ad7929cf964a520500c21e3',
 'name': 'The Senator Restaurant',
 'contact': {'phone': '4163647517',
  'formattedPhone': '(416) 364-7517',
  'twitter': 'thesenatorto',
  'facebook': '566290650069193',
  'facebookUsername': 'thesenatorto',
  'facebookName': 'The Senator'},
 'location': {'address': '249 Victoria Street',
  'crossStreet': 'btwn Dundas St E and Shuter St',
  'lat': 43.65564091455335,
  'lng': -79.37919882575557,
  'labeledLatLngs': [{'label': 'display',
    'lat': 43.65564091455335,
    'lng': -79.37919882575557}],
  'postalCode': 'M5B 1T8',
  'cc': 'CA',
  'city': 'Toronto',
  'state': 'ON',
  'country': 'Canada',
  'formattedAddress': ['249 Victoria Street (btwn Dundas St E and Shuter St)',
   'Toronto ON M5B 1T8',
   'Canada']},
 'canonicalUrl': 'https://foursquare.com/v/the-senator-restaurant/4ad7929cf964a520500c21e3',
 'categories': [{'id': '4bf58dd8d48988d147941735',
   'name': 'Diner',
   'pluralName': 'Diners',
   'shortName': 'Diner',
   'icon': {'prefix': 'h

In [19]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

7.6


In [20]:
try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')    

{'tier': 2, 'message': 'Moderate', 'currency': '$'}


In [21]:
# Now we leverage the Foursquare location data to the fullest
# let's get the Rating, Price, Tier and Currency for other restaurants


venue_id = '4c6abb8c2c29d13ad2720e41' # Subway
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.') 
    
    
venue_id = '4fbbb3ace4b01e39077b4d55' # Akashiro Japanese Restaurant & Bar
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.') 
    
        
venue_id = '4edbccaaf9abeea3db5268df' # Terrace Restaurant at Elmwood
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')
    
    
venue_id = '53ae6660498e1ac3c1282743' # Castries Bar & Five Way Restaurant
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')

    
venue_id = '56dd9d68498eb4e5edcb30f9' #  Spring Rolls | Japanese Restaurant in Toronto
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')
    
    
venue_id = '5165c333e4b07a7ad88d8a69' #   Joe's Indian Restaurant 
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')
    
    
venue_id = '4afcc52df964a520b82522e3' #  Donatello Restaurant 
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')
    
    
    
venue_id = '4ae9ef82f964a520cbb721e3' #  Alio Restaurant & Wine Bar
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.') 
    
    
venue_id = '4ad4c05ff964a52048f720e' # Hemispheres Restaurant & Bistro
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')
    
    
       
venue_id = '4b2027b5f964a520f82d24e3' #  Hong Shing Chinese Restaurant 
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.') 
    

5.6
{'tier': 1, 'message': 'Cheap', 'currency': '$'}
5.6
{'tier': 2, 'message': 'Moderate', 'currency': '$'}
This venue has not been rated yet.
{'tier': 2, 'message': 'Moderate', 'currency': '$'}
This venue has not been rated yet.
{'tier': 2, 'message': 'Moderate', 'currency': '$'}
This venue has not been rated yet.
No Details.
This venue has not been rated yet.
{'tier': 2, 'message': 'Moderate', 'currency': '$'}
7.3
{'tier': 3, 'message': 'Expensive', 'currency': '$'}
5.3
{'tier': 4, 'message': 'Very Expensive', 'currency': '$'}
This venue has not been rated yet.
No Details.
6.6
{'tier': 2, 'message': 'Moderate', 'currency': '$'}


##### Dataset Results above were extracted to csv format

In [ ]:
# Let's get the csv file to create new dataframe
df_dtres = pd.read_csv('DT_Res_final.csv')
df_dtres

##### New Dataframe: Rating and Price

In [ ]:
# Look for word 'Cheap'
df_word=df_dtres[df_dtres['Price'].str.contains('Cheap')].reset_index(drop=True)

# Hide the index
df_word.style.hide_index()

In [ ]:
# Look for word 'Moderate'
df_word=df_dtres[df_dtres['Price'].str.contains('Moderate')].reset_index(drop=True)

# Hide the index
df_word.style.hide_index()